R/get_ss3_exe.r

```diff
@@ -0,0 +1,85 @@
+#' Download the Stock Synthesis (SS3) executable
+#'
+#' Downloads the SS3 executable according to specified version and the user 
+#' operating system.
+#'
+#' @param dir The directory that you would like the executable downloaded to.
+#' @param version A character string of the executable version tag to download 
+#' (e.g.'v3.30.20' or 'v3.30.18').
+#' @return A string of the full file path to the downloaded executable
+#' @author Elizabeth F. Gugliotti
+#' @export
+#' @import gh
+#' @examples
+#' \dontrun{
+#' get_ss3_exe()
+#' get_ss3_exe(version = 'v3.30.18')
+#' }
+#' @description The `get_ss3_exe()` function uses the {gh} package to get either 
+#' the latest release (if version = NULL) or the specified version of the Stock 
+#' Synthesis executable for the appropriate operating system to the directory `dir` 
+#' (if dir = NULL, then the executable is downloaded to the working directory). 
+#' To view the version tags available go to 
+#' https://github.com/nmfs-stock-synthesis/stock-synthesis/tags
+
+get_ss3_exe <- function(dir = NULL, version = NULL){
+  # Get latest release if version not specified
+  if(is.null(version)){
+    latest_release <- gh::gh("GET /repos/nmfs-stock-synthesis/stock-synthesis/releases/latest", page = 1)
+    tag <- latest_release$tag_name 
+  } else {
+    # Otherwise get specified version
+    all_tags <- gh::gh("GET /repos/nmfs-stock-synthesis/stock-synthesis/tags")
+    df_tags <- as.data.frame(do.call(rbind, all_tags))
+    tags <- unlist(df_tags$name)
+    
+    if(!version %in% tags){
+      warning("The version tag that you entered is invalid or not in the right format, 
+              please go to https://github.com/nmfs-stock-synthesis/stock-synthesis/tags 
+              to get a correct version tag or version tag format")
+    } else{
+      tag <- version
+    }
+  }
+  
+  if(is.null(dir)){
```

Comment:
The function crashes if you try to download to a directory which doesn't exist. I would either add something like
```
  if (!dir.exists(dir)) {
    stop("Directory doesn't exist: ", dir)
  }
```
Or better yet, create the directory. There are various, surely inconsistent approaches to that in other r4ss functions, including this one in `copy_SS_inputs()` where you get to choose whether to allow it to create a new directory, and if so, whether to recursively created nested directories beyond just a single addition:
https://github.com/r4ss/r4ss/blob/main/R/copy_SS_inputs.R#L63-L70.

Reply:
going with the stop if directory doesn't exist. I'd rather people be downloading to a known place so it's more easily findable.